### ISSUE TITLE
Start search (find in page) at current scroll position instead of top

### ISSUE THREAD

**Issue by drlight-code** [qutebrowser/qutebrowser#1222]
I noticed on thing regarding incremental search on a page. The hits are actually always searched in order, even if i scroll further down in between. I would prefer the behavior of e.g. less which will continue its search from the top line where I'm scrolled to.

I do this for For example on this page http://www.juce.com/doc/classes when i search for something it would be really convenient to intermix spacebar for scrolling and n for next hit in order to skip parts of the page (alphabetic list) where I know what I'm searching for is not anyways.

We probably want to wait for #666 before implementing this twice.

This should be configurable by an option because people might prefer the old behavior.

**Comment by The-Compiler** [comment]
I'd prefer to implement it for QtWebKit too, if it's possible (I suspect it's not) - the QtWebKit backend will probably stay there for the forseeable future, i.e. at least as long Debian/Ubuntu don't package it.

**Comment by The-Compiler** [comment]
I took a quick look at this, and with both QtWebKit and QtWebEngine I don't see a straightforward way to implement it.

I opened [QTBUG-50420](https://bugreports.qt.io/browse/QTBUG-50420) with some suggestions for a better `QWebEnginePage::findText`, and also added this there.

**Comment by WloHu** [comment]
The requested behavior will be consistent with Vim/less and also to how I'm used to do it with VimFx.

I noticed 2 issues here:
- `search-next` and `search-prev` go to next/previous search text ignoring the scroll position;
- re-searching **ANY** text starts from previous match location instead from start of the page or first visible element (`/the<Enter>2n/the. To make it "reset" and search from the top some text that isn't on page should be searched.

Also I find this kind of annoying when my scroll position gets moved when I try to search for text which turns out to not be on the page before I confirm the search with `<Enter>`. Is there a setting to change that? If not should I open new issue?

**Comment by WloHu** [comment]
> The reason you're probably seeing this is that the first char of the search matches, scrolling you up.
Exactly. `:set search.incremental true` solves the issue as the default value is `true`.

Thanks for linking regex search issue.

As for search logic and regex it seems that there is no easy way. I totally agree that the best thing is to have those features built into web engine. The "dirty" solutions like JS scripts can be left out for qutebrowser users which would be easier to do when plugins are available.

**Comment by The-Compiler** [comment]
Here's a related Chromium issue: [128785 - Find in page should not scroll (to the top of the page)](https://bugs.chromium.org/p/chromium/issues/detail?id=128785)

Apparently it's possible to highlight some text at the beginning of the viewport, then searching will continue from there instead of the top of the page. It's a bit of a hack, but maybe we could do that?

**Comment by The-Compiler** [comment]
Merging this into #5652, as there's no other option as long as Chromium/QtWebEngine don't expose a way to do this.

**Comment by The-Compiler** [comment]
I'd still like to explore the "select some text at the beginning of the viewport" idea a bit more. Perhaps we can use the existing caret code to place the cursor there and it'll just work.
